core/linux-dist/x11.cpp

```diff
@@ -117,24 +136,35 @@ void input_x11_handle()
 
 void input_x11_init()
 {
-	x11_keymap[113] = DC_DPAD_LEFT;
-	x11_keymap[114] = DC_DPAD_RIGHT;
+	x11_keymap[KEY_LEFT] = DC_DPAD_LEFT;
+	x11_keymap[KEY_RIGHT] = DC_DPAD_RIGHT;
+
+	x11_keymap[KEY_UP] = DC_DPAD_UP;
+	x11_keymap[KEY_DOWN] = DC_DPAD_DOWN;
 
-	x11_keymap[111] = DC_DPAD_UP;
-	x11_keymap[116] = DC_DPAD_DOWN;
+	// Layout on a real DC controller
+	//   Y
+	// X   B
+	//   A
+	x11_keymap[KEY_S] = DC_BTN_X;
+	x11_keymap[KEY_X] = DC_BTN_A;
+	x11_keymap[KEY_D] = DC_BTN_Y;
+	x11_keymap[KEY_C] = DC_BTN_B;
 
-	x11_keymap[53] = DC_BTN_X;
-	x11_keymap[54] = DC_BTN_B;
-	x11_keymap[55] = DC_BTN_A;
+	// ???
```

Comment:
These were missing buttons on the original controller. They were used by some "arcade" controllers that were released by ascii and such iirc.